Update error notice for card errors

oikeme opened the pull request:
Update the error handling to account for a cvc error response potentially being an invalid card expiration.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #

## Proposed Changes

* Update the error notice in the card additions page to indicate when a received CVC error may actually be for an incorrect expiration date and account for both in the erro messages.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Use the Jetpack Cloud Live link below to access the payments add page at `partner-portal/payment-methods/add`
* Enter in a card number with an incorrect CVC. Ensure that the new error message is displayed. 
* Enter in a card number with a correct CVC but incorrect expiration date. Ensure that an appropriate error is displayed. 

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

oikeme committed:
Update error notice for card errors

Update the error handling to account for a cvc error response potentially being an invalid card expiration.

oikeme committed:
set tranactionError

set transaction error when CVC is included in the error message

yashwin commented on client/jetpack-cloud/sections/partner-portal/payment-method-form/index.tsx:
```diff
@@ -140,6 +140,9 @@ function PaymentMethodForm() {
 
 	const handleChangeError = useCallback(
 		( { transactionError }: { transactionError: string | null } ) => {
+			if ( !! transactionError && transactionError.includes( 'CVC' ) ) {
```
We should be doing `transactionError.includes( 'cvc' )`

```suggestion
			if ( transactionError && transactionError.includes( 'cvc' ) ) {
```

yashwin commented on client/jetpack-cloud/sections/partner-portal/payment-method-form/index.tsx:
```diff
@@ -140,6 +140,9 @@ function PaymentMethodForm() {
 
 	const handleChangeError = useCallback(
 		( { transactionError }: { transactionError: string | null } ) => {
+			if ( !! transactionError && transactionError.includes( 'CVC' ) ) {
```
We should be doing `transactionError.includes( 'cvc' )`

```suggestion
			if ( transactionError && transactionError.includes( 'cvc' ) ) {
```

oikeme replied:
Thanks for pointing this out. I've updated to include lower case checks for both error code situations discussed.

yashwin reviewed (changes requested):
Thanks for working on this. 

I have left a comment

oikeme committed:
Update payment method error check

Update the error message and conditions for generic cvc/expiration error code display.

oikeme committed:
Update transactionError check

Remove null value check on transactionError

oikeme committed:
Update substring check for the cvc code

Update check to include 'security code' substring to be more precise

yashwin reviewed (approved):
Thanks for addressing my comments and fixing this issue. 

LGTM!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/11019216

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @oikeme, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Your payment method cvc code or expiration date is invalid.`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/11019216).
